### PR TITLE
docs: Flux 101 per skyscrapers/platform#1187

### PR DESCRIPTION
**Pull request by simonrondelez**
# Description

This PR adds documentation on how to use Flux on the Skyscrapers Developer Platform.

also fixes some URLs

# Related Issue

skyscrapers/platform#1187

# Motivation and Context

New feature: Flux 🎊 

# How Has This Been Tested?

# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist

- [ ] I have followed the PR process as described [here](https://github.com/skyscrapers/documentation/blob/master/coding_guidelines/git.md#pull-requests)
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: kubernetes/flux.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        └── apps.yaml #(Kustomization to point to the apps folder/repo)
+k8s-clusters/
+├── production-cluster-name.yaml
+└── staging-cluster-name.yaml
+terraform/
+├── live/
+└── modules/
+```
+
+### Directory Breakdown
```
Comment (duboisph) [inline kubernetes/flux.md:58]: I'd add a note or similar to make clear that this is deployed in the `skyscrapers/customer_name` repo that we provide. Also that `flux/clusters/*` Is required to be in that repo, but the `flux/apps` can be split off to other repos (depending on the apps.yaml overlay we add).

File: kubernetes/flux.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+**flux/**: Directory to hold all Flux-related configuration files.
+
+- **apps/**: Optional directory to organize application configurations.
+  - **base/**: Base configurations for all environments.
+  - **production/**: Specific configurations for the production environment.
+  - **staging/**: Specific configurations for the staging environment.
+- **clusters/**: Directory for managing cluster-specific configurations.
+  - **production-cluster-name/**: Configuration for the production cluster.
+    - **flux-system/**: Contains the core Flux system configurations.
```
Comment (duboisph) [inline kubernetes/flux.md]:
```suggestion
    - **flux-system/**: Contains the core Flux system configurations. Automatically managed by SkS, don't edit!
```

File: kubernetes/flux.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      - **gotk-components.yaml**: Describes the components to be installed by Flux.
+      - **gotk-sync.yaml**: Configuration for syncing the repository with the cluster.
```
Comment (duboisph) [inline kubernetes/flux.md:70]: I'd just remove this part? Not directly relevant for the customer (and they should not touch).

```suggestion
```

File: kubernetes/flux.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      - **kustomization.yaml**: Kustomization file for managing resources.
```
Comment (duboisph) [inline kubernetes/flux.md:71]: I'd just remove this part? Not directly relevant for the customer (and they should not touch).

```suggestion
```

File: kubernetes/flux.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  ref:
+    branch: main
+  secretRef:
+    name: apps-deploykey
+  url: ssh://git@github.com/<organisation>/<repo-name>.git
+```
+
+## Example Configuration
+
+This section provides shows some examples of the configuration files used to manage Flux resources. A for more in depth explanation of the configuration files, please refer to the [Flux documentation](https://fluxcd.io/docs/) or reach out to us.
```
Comment (duboisph) [inline kubernetes/flux.md:94]:
```suggestion
This section provides some examples of the configuration files used to manage Flux resources. A for more in depth explanation of the configuration files, please refer to the [Flux documentation](https://fluxcd.io/docs/) or reach out to us.
```

**Comment by Vahan90** [comment]
Reviewed as per @simonrondelez [request](https://skyscrapers.slack.com/archives/C02PZSS4X/p1717077481124289?thread_ts=1716801040.293539&cid=C02PZSS4X), and it lgtm

File: kubernetes/flux.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
+# Flux
+
+## Introduction
+
+[Flux](https://fluxcd.io/) is a way to deploy and maintain your applications and components through GitOps. It is designed to keep your Kubernetes clusters in sync based on the configuration in git and to automate updates to configuration when Flux detects it. This documentation provides guidance on setting up and managing your repository structure using Flux in the cooperation with Skyscrapers.
```
Comment (duboisph) [inline kubernetes/flux.md:5]:
```suggestion
[Flux](https://fluxcd.io/) is a way to deploy and maintain your applications and components through [GitOps](https://www.gitops.tech/#what-is-gitops). It is designed to keep your Kubernetes clusters in sync based on the configuration in git and to automate updates to configuration when Flux detects it. This documentation provides guidance on setting up and managing your repository structure using Flux in the cooperation with Skyscrapers.
```

Comment (duboisph) [reply kubernetes/flux.md:5]: Maybe add something like this (what is the customer benefit): 

In short, this means flux will pull your changes from Git and keep everything reconciled. For example if you commit a neww container image to your helm chart, flux will detect this and perform a helm upgrade directly from within the Kubernetes cluster.

File: kubernetes/flux.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+  - [Repository Structure](#repository-structure)
+    - [Directory Breakdown](#directory-breakdown)
+    - [Applications](#applications)
+  - [Example Configuration](#example-configuration)
+    - [kustomization.yaml](#kustomizationyaml)
+    - [apps.yaml](#appsyaml)
+      - [Internal within the same repository](#internal-within-the-same-repository)
+      - [External repository](#external-repository)
+
+
```
Comment (duboisph) [inline kubernetes/flux.md:19]:
```suggestion

```

File: kubernetes/flux.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+**flux/**: Directory to hold all Flux-related configuration files.
+
+- **apps/**: Optional directory to organize application configurations.
+  - **base/**: Base configurations for all environments.
+  - **production/**: Specific configurations for the production environment.
+  - **staging/**: Specific configurations for the staging environment.
+- **clusters/**: Directory for managing cluster-specific configurations.
+  - **production-cluster-name/**: Configuration for the production cluster.
+    - **flux-system/**: Contains the core Flux system configurations. This is automatically provisioned for you by Skyscrapers/Flux and can't be modified manually.
+    - **apps.yaml**: Kustomization file pointing to the apps folder or repository. See [Applications section](#applications) for more information.
```
Comment (duboisph) [inline kubernetes/flux.md:65]:
```suggestion
    - **apps.yaml**: Kustomization overlay pointing to the apps folder or repository. See [Applications section](#applications) for more information.
```

**Review by duboisph — APPROVED** [review]

**Pull request merged**